### PR TITLE
FOLSPRINGB-21 Add CQL support

### PR DESCRIPTION
**Pull request by psmagin**
## Purpose
Support CQL-to-SQL queries transformation
https://issues.folio.org/browse/FOLSPRINGB-21

## Approach
* `JpaCqlRepository` was created. It extends `JpaRepository` with methods that support CQL-queries
* Implementation of the repository interface was created
* `JpaCqlRepositoryFactoryBean` was created to tell Spring to use suitable implementation of the repository interface
* ` JpaCqlConfiguration` was created to autoconfigure repositories creation

### CODE REVIEW

File: src/test/java/org/folio/spring/cql/Cql2JpaCriteriaTest.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+import lombok.Getter;
+import org.hibernate.query.Query;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+import org.folio.spring.cql.domain.Person;
+
+class Cql2JpaCriteriaTest {
```
Comment (dmtkachenko) [inline src/test/java/org/folio/spring/cql/Cql2JpaCriteriaTest.java:17]: 👍

File: src/main/java/org/folio/spring/data/OffsetRequest.java [hunk 2]
```diff
@@ -38,6 +37,14 @@ public OffsetRequest(long offset, int limit, Sort sort) {
     this.sort = Objects.requireNonNull(sort, "Sorting cannot be null. Use Sort.unsorted() instead");
   }
 
+  public static OffsetRequest of(int offset, int limit) {
+    return of(offset, limit, Sort.unsorted());
```
Comment (dmtkachenko) [inline src/main/java/org/folio/spring/data/OffsetRequest.java:41]: it's better to keep it in consistence with constructor: use DEFAULT_SORT

Comment (psmagin) [reply src/main/java/org/folio/spring/data/OffsetRequest.java]: Thanks! Fixed it.

**Review by dmtkachenko — APPROVED** [review]

File: src/main/java/org/folio/spring/cql/Cql2JpaCriteria.java [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+    boolean isString = String.class.equals(field.getJavaType());
+
+    String comparator = node.getRelation().getBase().toLowerCase();
+
+    switch (comparator) {
+      case "=":
+        if (CqlTermFormat.NUMBER.equals(modifiers.getCqlTermFormat())) {
+          return queryBySql(field, node, comparator, cb);
+        }
+      case "adj":
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/spring/cql/Cql2JpaCriteria.java:230]: Can we use enum for all this things "=", "adj", "all", "any", "==" and so on ?

**Review by Igor-Gorchakov — APPROVED** [review]

**Pull request merged**